### DISCUSSION TITLE
process.env.* is undefined when sites crawled by indexing

### DISCUSSION THREAD

**Discussion by ReeceM** [vercel/vercel#5657] [Help] [answered]
I am not sure if this is the right repo \:)

I have a simple section of Schema markup for a website,

the code is the following:

```javascript
import {SITE_URL} from '../constants.js'
....
        <script
          type='application/ld+json'
          dangerouslySetInnerHTML={{
            __html: JSON.stringify({
              "@context": "https://schema.org",
              "@type": "Organization",
              "url": `${SITE_URL}`,
              "logo": `${SITE_URL}/favicon/maskable_icon.png`
            })
          }}
        />
```

The constants file has: 
```javascript
export const SITE_URL = process.env.SITE_URL;
```

But when crawled by google, it says that there is an error, looking at 'their rendered' HTML, the value that should be `SITE_URL` is undefined,

```html
    <script type="application/ld+json">
        {
            "@context": "https://schema.org",
            "@type": "Organization",
            "url": "undefined",
            "logo": "undefined/favicon/maskable_icon.png"
        }
    </script>
```

BUT, if you make a cURL request, or even `view-source:xyz` with chrome, you get the following:

```html
    <script type="application/ld+json">
        {
            "@context": "https://schema.org",
            "@type": "Organization",
            "url": "https://redacted-for-sanity",
            "logo": "https://redacted-for-sanity/favicon/maskable_icon.png"
        }
    </script>
```

So I am not sure what is really happening, to me it appears that for some reason the env variables set in the dashboard are not being used in some instances or getting different values passed to it?!

Or are there instances where this code is not available ?

Any help or direction of where to look would be greatly appreciated :)

**Comment by rohanbhatia96** [comment, marked as answer]
For the env variable to be availble to crawlers, you need to prepend the variable name with NEXT_PUBLIC.
In your case, if you change the env variable name to NEXT_PUBLIC_SITE_URL, it should work
